### PR TITLE
[Hub Generated] Review request for Microsoft.StorageCache to add version stable/2023-05-01

### PR DESCRIPTION
**Pull request by tibanyas**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/tibanyas-storagecache-Microsoft.StorageCache-2023-05-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in #19724

**Comment by tibanyas** [comment]
This PR is for the new 2023-05-01 stable API version for Microsoft.StorageCache and is based off of the Approved 2023-03-01-preview API version. 

The most recent stable API version is 2023-01-01, and the main difference between 2023-03-01-preview and 2023-01-01 is the support for the amlFilesystem resource, which went public preview with the 2023-03-01-preview API version.  The amlFilesystem resource is planning to go GA with the 2023-05-01 API version and this is the main difference with the most recent stable API version (2023-01-01).

**After creating 2023-05-01, based off of 2023-03-01-preview, the following changes were made:**

**1. Remove reference to scalingFactor and scalableSkus from storageCache.json and associated examples.**
Scalable Skus is a preview feature for storagecache and it is not going GA at this time.  Therefore, these fields were in the last preview API version, but needed removed for this stable version.

**2. Added a new clientInfo structure in amlfilesystem.json.**
- Previous API versions had three client read-only fields under the AmlFilesystem structure: mgsAddress, mountCommand, and lustreVersion. 
- We are adding three new client fields in this new API version: (persistentVolumeClaim, persistentVolume, and storageClass)
- Rather than continuing to expand the client fields under the AmlFilesystem structure, we decided to add a reference to a new clientInfo structure instead, and move the client read only fields under there. All six client fields (3 existing fields, 3 new fields) were added to the new clientInfo structure.
- The 3 existing client fields were removed from the AmlFilesystem structure and moved inside of the clientInfo structure instead, and the 3 new client fields were added inside the new clientInfo structure.

**Comment by tibanyas** [comment]
The Swagger LintDiff errors were all present and noted in our previous review for 2023-03-01-preview; see this PR:
https://github.com/Azure/azure-rest-api-specs/pull/19724

They were all noted and approved at that time.  Nothing is new here related to LintDiff

**Comment by tadelesh** [comment]
@tibanyas do your service has plan to resolve all the lint error? We bypassed the lint error based on the promise that service will handle them in next version.

**Comment by tadelesh** [comment]
SDK breaking changes is possibly from m4 bug. Need to handle when SDK release. @MaryGao

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-05-01/amlfilesystem.json [hunk 1]
```diff
@@ -0,0 +1,1145 @@
...
+          "type": "object",
+          "required": [
+            "filesystemSubnet",
+            "storageCapacityTiB",
+            "maintenanceWindow"
+          ]
+        }
+      }
+    },
+    "AmlFilesystemUpdate": {
```
Comment (sjanamma) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-05-01/amlfilesystem.json:746]: >"AmlFilesystemUp [](http://example.com/codeflow?start=4&length=16)

Track resource update must be able to update sku long with tags

**Comment by sjanamma** [comment]
> The Swagger LintDiff errors were all present and noted in our previous review for 2023-03-01-preview; see this PR: #19724
> 
> They were all noted and approved at that time. Nothing is new here related to LintDiff

You are introducing a new version and breaking changes have been approved. This is the good opportunity for you to address the LintDiff issues

**Review by sjanamma — APPROVED** [review]
:shipit:

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-05-01/amlfilesystem.json (hunk 1 above)
Comment (tibanyas) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-05-01/amlfilesystem.json:746]: @sjanamma ,
A SKU cannot be modified in our service; this is intentional.  We have it's mutability classified accordingly as shown below:

"SkuName": {
     "x-ms-mutability": [
       "read",
       "create"
     ],

**Comment by tibanyas** [comment]
@sjanamma ,

Regarding the LintDiff errors, 

The following errors/warnings are alarms:
- LongRunningResponseStatusCode
- AvoidNestedProperties
- AvoidNextedProperties.

The other two errors listed are:

- PatchSku
- PatchIdentity.  

As is the case with Sku, our service doesn't support modifying the Identity object, and we have its mutability classified accordingly as shown below: 

"AmlFilesystemIdentity": {
      "type": "object",
      "description": "Managed Identity properties.",
      "x-ms-mutability": [
        "read",
        "create"
      ],

Thanks,
Tim

**Pull request merged**